airflow/models/skipmixin.py

```diff
@@ -120,23 +74,11 @@ def skip(
             return
 
         task_ids_list = [d.task_id for d in task_list]
-        SkipMixin._set_state_to_skipped(dag_id, run_id, task_ids_list, session)
```

Comment:
I wonder if we even need to preserve this functionality - it surely won't work when there's no direct access to the DB from the operator  + overall it looks like a short cut to set tasks to skipped before `NotPreviouslySkippedDep` handles it (breaks single responsibility?).
If we decide to preserve it, we'll need to signal the task runner with exceptions.

Reply:
The TaskSDK could make an API call dedicated to handling skipping if we want.

Reply:
And yeah, I think we should do that. The TIDep skipping again is I think do catch it when things get cleared but the previous skip result is maintained -- see #7276, but we should still skip tasks immediately in the API server handling path.